pkg/querybuilder/where_clause_visitor.go

```diff
@@ -638,28 +640,38 @@ func (v *filterExpressionVisitor) VisitFunctionCall(ctx *grammar.FunctionCallCon
 	for _, key := range keys {
 		var fieldName string
 
-		if strings.HasPrefix(key.Name, v.jsonBodyPrefix) {
-			fieldName, _ = v.jsonKeyToKey(context.Background(), key, qbtypes.FilterOperatorUnknown, value)
+		if functionName == "hasToken" {
+			// this will only work with string.
+			if _, ok := value[0].(string); !ok {
+				v.errors = append(v.errors, fmt.Sprintf("function `%s` expects value parameter to be a string", functionName))
+				return ""
+			}
+			conds = append(conds, fmt.Sprintf("hasToken(LOWER(%s), LOWER(%s))", key.Name, v.builder.Var(value[0])))
```

Comment:
This can't be a general `key.Name` as it will fail for any key that is not a direct column, such as `hasToken(addlInfo.message, 'XYX')`. Either allow only body or handle all fields in general.

Reply:
Makes sense, let me enable it just for body now.